test/CMakeLists.txt

```diff
@@ -0,0 +1,234 @@
+include(FetchContent)
+
+# Download and unpack googletest at configure time
+configure_file(CMakeLists-gtest.txt.in googletest-download/CMakeLists.txt)
```

Comment:
This should be done with fetchContent but it would require the cmake to be upgraded on all docker images first.